src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,64 @@ export class CodeActionModel extends Disposable {
 					return;
 				}
 
-				const actions = createCancelablePromise(token => getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token));
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._showInlineQuickfxies()) {
+						const codeActionSet = await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
+
+						// Search for quickfixes in the curret code action set.
+						const foundQuickfix = codeActionSet.validActions?.some(action => action.action.kind === 'quickfix') || false;
+
+						const allMarkers = this._markerService.read({ resource: model.uri });
+
+						// If markers exists, and there are no quickfixes found or length is zero, check for quickfixes on that line.
+						if (allMarkers.length > 0 && (!foundQuickfix || codeActionSet.validActions.length === 0) && trigger.trigger.type === CodeActionTriggerType.Invoke) {
+							const currPosition: Position = trigger.selection.getPosition();
+							let trackedPosition: Position = currPosition;
+							let distance = Number.MAX_VALUE;
+							let toBeModified = false;
+
+							allMarkers.forEach((marker: IMarker) => {
+								const col = marker.endColumn;
+								const row = marker.endLineNumber;
+
+								// Found quickfix on the same line and check relative distance to other markers
+								if (row === currPosition.lineNumber && Math.abs(currPosition.column - col) < distance) {
```

Comment:
What if the diagnostic is multiple lines? Should we also check the start position for cases like:

```
text cursor text diagStart
text diagEnd
```